
        <!DOCTYPE html>
        <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1.0">
            <title>Automated Test Generation - Generated by Lance</title>
            <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;700&display=swap" rel="stylesheet">
            <link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.23.0/themes/prism-okaidia.min.css" rel="stylesheet" />
            <style>                
                body {
                    font-family: 'Roboto', sans-serif;
                    background-color: #f8f9fa;
                    color: #333;
                    margin: 0;
                    padding: 20px;
                    }
                    header {
                        margin-bottom: 20px;
                        text-align: center;
                    }
                    header h1 {
                        margin: 0;
                        font-size: 2.5em;
                        color: #007bff;
                    }
                    header p {
                        margin: 5px 0 0;
                        font-size: 1.2em;
                        color: #6c757d;
                    }
                    table {
                        width: 100%;
                        border-collapse: collapse;
                        box-shadow: 0 2px 3px rgba(0,0,0,0.1);
                        background-color: #fff;
                    }
                    th, td {
                        padding: 15px;
                        text-align: left;
                        border: 1px solid #dee2e6;
                    }
                    th {
                        background-color: #007bff;
                        color: #fff;
                        font-weight: 700;
                    }
                    tr:nth-child(even) {
                        background-color: #f2f2f2;
                    }
                    .status-pass {
                        color: green;
                        font-weight: 700;
                    }
                    .status-fail {
                        color: red;
                        font-weight: 700;
                    }
                    pre {
                        background-color: #282a36 !important;
                        color: #f8f8f2 !important;
                        padding: 10px;
                        border-radius: 5px;
                        overflow: auto;
                    }
                    code {
                        font-family: 'Courier New', Courier, monospace;
                    }
            </style>
        </head>
        <body>
            <header>
                <h1>Test Results</h1>
                <p>Generated by Lance</p>
            </header>
            
            <table>
                <tr>
                    <th>Status</th>
                    <th>Label</th>
                    <th>Reason</th>
                    <th>Exit Code</th>
                    <th>Line Coverage</th>
                    <th>Branch Coverage</th>
                    <th>Stderr</th>
                    <th>Stdout</th>
                    <th>Test</th>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_0</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullChecker(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.036 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that GaussNewtonOptimizer constructor with a ConvergenceChecker throws a NullArgumentException when the checker is null.\n', 'test_name': 'testConstructorNullChecker', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullChecker() {\n    new GaussNewtonOptimizer((ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_0</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullCheckerBoolean(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.036 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that GaussNewtonOptimizer constructor with a boolean and a ConvergenceChecker throws a NullArgumentException when the checker is null.\n', 'test_name': 'testConstructorNullCheckerBoolean', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullCheckerBoolean() {\n    new GaussNewtonOptimizer(true, (ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_0</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeSingular(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.012 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.ConvergenceException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that doOptimize method throws a ConvergenceException when the normal equations are singular.\n', 'test_name': 'testDoOptimizeSingular', 'test_code': '@Test(expected = ConvergenceException.class)\npublic void testDoOptimizeSingular() {\n    // implement a test that covers the singular case\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_0</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeInternalError(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.011 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.MathInternalError
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that doOptimize method throws a MathInternalError when an internal error occurs.\n', 'test_name': 'testDoOptimizeInternalError', 'test_code': '@Test(expected = MathInternalError.class)\npublic void testDoOptimizeInternalError() {\n    // implement a test that covers the internal error case\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_0</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testCheckParametersBounds(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.009 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.MathUnsupportedOperationException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that checkParameters method throws a MathUnsupportedOperationException when bounds are passed.\n', 'test_name': 'testCheckParametersBounds', 'test_code': '@Test(expected = MathUnsupportedOperationException.class)\npublic void testCheckParametersBounds() {\n    // implement a test that covers the bounds case\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-INFO">INFO</td>
                    <td>g_0</td>
                    <td>40.61068248748779</td>
                    <td>3346</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullChecker(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.029 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the constructor throws NullArgumentException when the checker is null.', 'test_name': 'testConstructorNullChecker', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullChecker() {\n    new GaussNewtonOptimizer((ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.NullArgumentException;\nimport org.apache.commons.math3.optim.ConvergenceChecker;\nimport org.apache.commons.math3.optim.PointVectorValuePair;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullCheckerBoolean(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.03 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the constructor throws NullArgumentException when the checker is null and useLU is specified.', 'test_name': 'testConstructorNullCheckerBoolean', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullCheckerBoolean() {\n    new GaussNewtonOptimizer(true, (ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeSingular(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.034 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.ConvergenceException> but was<java.lang.NullPointerException>
	at org.apache.commons.math3.optim.nonlinear.vector.MultivariateVectorOptimizer.getTarget(MultivariateVectorOptimizer.java:106)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer.doOptimize(GaussNewtonOptimizer.java:91)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeSingular(GaussNewtonOptimizerTest.java:49)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the doOptimize method throws ConvergenceException when the problem is singular.', 'test_name': 'testDoOptimizeSingular', 'test_code': '@Test(expected = ConvergenceException.class)\npublic void testDoOptimizeSingular() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    optimizer.doOptimize();\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeInternalError(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.029 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.MathInternalError> but was<java.lang.NullPointerException>
	at org.apache.commons.math3.optim.nonlinear.vector.MultivariateVectorOptimizer.getTarget(MultivariateVectorOptimizer.java:106)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer.doOptimize(GaussNewtonOptimizer.java:91)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeInternalError(GaussNewtonOptimizerTest.java:49)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the doOptimize method throws MathInternalError when an internal error occurs.', 'test_name': 'testDoOptimizeInternalError', 'test_code': '@Test(expected = MathInternalError.class)\npublic void testDoOptimizeInternalError() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            throw new MathInternalError();\n        }\n    });\n    optimizer.doOptimize();\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_1</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[49,18] checkParameters() has private access in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the checkParameters method throws MathUnsupportedOperationException when bounds are specified.', 'test_name': 'testCheckParametersBounds', 'test_code': '@Test(expected = MathUnsupportedOperationException.class)\npublic void testCheckParametersBounds() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    optimizer.checkParameters();\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullChecker(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.025 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the constructor with a null convergence checker.', 'test_name': 'testConstructorNullChecker', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullChecker() {\n    new GaussNewtonOptimizer((ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.NullArgumentException;\nimport org.apache.commons.math3.optim.ConvergenceChecker;\nimport org.apache.commons.math3.optim.PointVectorValuePair;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullCheckerBoolean(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.025 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the constructor with a boolean and a null convergence checker.', 'test_name': 'testConstructorNullCheckerBoolean', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullCheckerBoolean() {\n    new GaussNewtonOptimizer(true, (ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_2</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[49,18] cannot find symbol
  symbol:   method setTarget(double[])
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[50,18] cannot find symbol
  symbol:   method setStartPoint(double[])
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the doOptimize method with a singular matrix.', 'test_name': 'testDoOptimizeSingular', 'test_code': '@Test(expected = ConvergenceException.class)\npublic void testDoOptimizeSingular() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    optimizer.setTarget(new double[1]);\n    optimizer.setStartPoint(new double[1]);\n    optimizer.doOptimize();\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_2</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[49,18] cannot find symbol
  symbol:   method setTarget(double[])
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[50,18] cannot find symbol
  symbol:   method setStartPoint(double[])
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the doOptimize method with an internal error.', 'test_name': 'testDoOptimizeInternalError', 'test_code': '@Test(expected = MathInternalError.class)\npublic void testDoOptimizeInternalError() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            throw new MathInternalError();\n        }\n    });\n    optimizer.setTarget(new double[1]);\n    optimizer.setStartPoint(new double[1]);\n    optimizer.doOptimize();\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_2</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[49,18] cannot find symbol
  symbol:   method setLowerBound(double[])
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[50,18] cannot find symbol
  symbol:   method setUpperBound(double[])
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the checkParameters method with bounds.', 'test_name': 'testCheckParametersBounds', 'test_code': '@Test(expected = MathUnsupportedOperationException.class)\npublic void testCheckParametersBounds() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    optimizer.setLowerBound(new double[1]);\n    optimizer.setUpperBound(new double[1]);\n    try {\n        optimizer.doOptimize();\n    } catch (MathUnsupportedOperationException e) {\n        assertEquals(LocalizedFormats.CONSTRAINT, e.getMessage());\n        throw e;\n    }\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_3</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[48,13] cannot find symbol
  symbol:   variable Assert
  location: class org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the constructor with a null convergence checker', 'test_name': 'testConstructorNullChecker', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullChecker() {\n    try {\n        new GaussNewtonOptimizer((ConvergenceChecker<PointVectorValuePair>) null);\n        Assert.fail("expected NullArgumentException");\n    } catch (NullArgumentException e) {\n        // expected\n    }\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.NullArgumentException;\nimport org.apache.commons.math3.optim.ConvergenceChecker;\nimport org.apache.commons.math3.optim.PointVectorValuePair;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_3</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[45,13] cannot find symbol
  symbol:   variable Assert
  location: class org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the constructor with a null convergence checker and a boolean flag', 'test_name': 'testConstructorNullCheckerBoolean', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullCheckerBoolean() {\n    try {\n        new GaussNewtonOptimizer(true, (ConvergenceChecker<PointVectorValuePair>) null);\n        Assert.fail("expected NullArgumentException");\n    } catch (NullArgumentException e) {\n        // expected\n    }\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_3</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeSingular(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.03 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.ConvergenceException> but was<java.lang.NullPointerException>
	at org.apache.commons.math3.optim.nonlinear.vector.MultivariateVectorOptimizer.getTarget(MultivariateVectorOptimizer.java:106)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer.doOptimize(GaussNewtonOptimizer.java:91)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeSingular(GaussNewtonOptimizerTest.java:53)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the doOptimize method with a singular matrix', 'test_name': 'testDoOptimizeSingular', 'test_code': '@Test(expected = ConvergenceException.class)\npublic void testDoOptimizeSingular() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    // set target and start point\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    // mock the computeObjectiveValue and computeResiduals methods\n    optimizer.doOptimize();\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_3</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[55,13] cannot find symbol
  symbol:   variable Assert
  location: class org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the doOptimize method with an internal error', 'test_name': 'testDoOptimizeInternalError', 'test_code': '@Test(expected = MathInternalError.class)\npublic void testDoOptimizeInternalError() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            throw new MathInternalError();\n        }\n    });\n    // set target and start point\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    // mock the computeObjectiveValue and computeResiduals methods\n    try {\n        optimizer.doOptimize();\n        Assert.fail("expected MathInternalError");\n    } catch (MathInternalError e) {\n        // expected\n    }\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_0_3</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[54,13] cannot find symbol
  symbol:   variable Assert
  location: class org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the checkParameters method with bounds', 'test_name': 'testCheckParametersBounds', 'test_code': '@Test(expected = MathUnsupportedOperationException.class)\npublic void testCheckParametersBounds() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    // set lower and upper bounds\n    double[] lowerBound = new double[1];\n    double[] upperBound = new double[1];\n    try {\n        optimizer.doOptimize();\n        Assert.fail("expected MathUnsupportedOperationException");\n    } catch (MathUnsupportedOperationException e) {\n        // expected\n    }\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-INFO">INFO</td>
                    <td>f_0</td>
                    <td>144.28174924850464</td>
                    <td>18510</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullChecker(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.037 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the GaussNewtonOptimizer constructor throws a NullArgumentException when the convergence checker is null.\n', 'test_name': 'testConstructorNullChecker\n', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullChecker() {\n    new GaussNewtonOptimizer((ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.NullArgumentException;\nimport org.apache.commons.math3.optim.ConvergenceChecker;\nimport org.apache.commons.math3.optim.PointVectorValuePair;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullCheckerBoolean(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.028 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the GaussNewtonOptimizer constructor throws a NullArgumentException when the convergence checker is null and useLU is true.\n', 'test_name': 'testConstructorNullCheckerBoolean\n', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullCheckerBoolean() {\n    new GaussNewtonOptimizer(true, (ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': ''}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeSingular(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.04 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.ConvergenceException> but was<java.lang.NullPointerException>
	at org.apache.commons.math3.optim.nonlinear.vector.MultivariateVectorOptimizer.getTarget(MultivariateVectorOptimizer.java:106)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer.doOptimize(GaussNewtonOptimizer.java:91)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeSingular(GaussNewtonOptimizerTest.java:53)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the doOptimize method throws a ConvergenceException when the linear problem is singular.\n', 'test_name': 'testDoOptimizeSingular\n', 'test_code': '@Test(expected = ConvergenceException.class)\npublic void testDoOptimizeSingular() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    // set target and start point\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    // mock the computeObjectiveValue and computeResiduals methods\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.ConvergenceException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeInternalError(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.036 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.MathInternalError> but was<java.lang.NullPointerException>
	at org.apache.commons.math3.optim.nonlinear.vector.MultivariateVectorOptimizer.getTarget(MultivariateVectorOptimizer.java:106)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer.doOptimize(GaussNewtonOptimizer.java:91)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeInternalError(GaussNewtonOptimizerTest.java:53)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the doOptimize method throws a MathInternalError when the convergence checker throws a MathInternalError.\n', 'test_name': 'testDoOptimizeInternalError\n', 'test_code': '@Test(expected = MathInternalError.class)\npublic void testDoOptimizeInternalError() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            throw new MathInternalError();\n        }\n    });\n    // set target and start point\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    // mock the computeObjectiveValue and computeResiduals methods\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathInternalError;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_1</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[52,18] checkParameters() has private access in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the checkParameters method throws a MathUnsupportedOperationException when bounds are set.\n', 'test_name': 'testCheckParametersBounds\n', 'test_code': '@Test(expected = MathUnsupportedOperationException.class)\npublic void testCheckParametersBounds() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    // set lower and upper bounds\n    double[] lowerBound = new double[1];\n    double[] upperBound = new double[1];\n    optimizer.checkParameters();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathUnsupportedOperationException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-INFO">INFO</td>
                    <td>g_1</td>
                    <td>43.903175354003906</td>
                    <td>4898</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullChecker(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.025 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the constructor with a null convergence checker', 'test_name': 'testConstructorNullChecker', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullChecker() {\n    new GaussNewtonOptimizer((ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.NullArgumentException;\nimport org.apache.commons.math3.optim.ConvergenceChecker;\nimport org.apache.commons.math3.optim.PointVectorValuePair;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullCheckerBoolean(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.028 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the constructor with a null convergence checker and a boolean flag', 'test_name': 'testConstructorNullCheckerBoolean', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullCheckerBoolean() {\n    new GaussNewtonOptimizer(true, (ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_1</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[51,18] cannot find symbol
  symbol:   method setTarget(double[])
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[52,18] cannot find symbol
  symbol:   method setStartPoint(double[])
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the doOptimize method with a singular matrix', 'test_name': 'testDoOptimizeSingular', 'test_code': '@Test(expected = ConvergenceException.class)\npublic void testDoOptimizeSingular() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    optimizer.setTarget(target);\n    optimizer.setStartPoint(startPoint);\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.ConvergenceException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_1</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[51,18] cannot find symbol
  symbol:   method setTarget(double[])
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[52,18] cannot find symbol
  symbol:   method setStartPoint(double[])
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the doOptimize method with an internal error', 'test_name': 'testDoOptimizeInternalError', 'test_code': '@Test(expected = MathInternalError.class)\npublic void testDoOptimizeInternalError() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            throw new MathInternalError();\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    optimizer.setTarget(target);\n    optimizer.setStartPoint(startPoint);\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathInternalError;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_1</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[51,18] cannot find symbol
  symbol:   method setLowerBound(double[])
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[52,18] cannot find symbol
  symbol:   method setUpperBound(double[])
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[53,18] checkParameters() has private access in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the checkParameters method with bounds', 'test_name': 'testCheckParametersBounds', 'test_code': '@Test(expected = MathUnsupportedOperationException.class)\npublic void testCheckParametersBounds() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] lowerBound = new double[1];\n    double[] upperBound = new double[1];\n    optimizer.setLowerBound(lowerBound);\n    optimizer.setUpperBound(upperBound);\n    optimizer.checkParameters();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathUnsupportedOperationException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullChecker(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.027 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that constructor throws exception when checker is null', 'test_name': 'testConstructorNullChecker', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullChecker() {\n    new GaussNewtonOptimizer((ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.NullArgumentException;\nimport org.apache.commons.math3.optim.ConvergenceChecker;\nimport org.apache.commons.math3.optim.PointVectorValuePair;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullCheckerBoolean(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.03 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that constructor throws exception when checker is null with useLU', 'test_name': 'testConstructorNullCheckerBoolean', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullCheckerBoolean() {\n    new GaussNewtonOptimizer(true, (ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_2</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[53,9] cannot find symbol
  symbol:   variable optimizertarget
  location: class org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[54,18] cannot find symbol
  symbol:   variable startPoint
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that doOptimize throws exception when matrix is singular', 'test_name': 'testDoOptimizeSingular', 'test_code': '@Test(expected = ConvergenceException.class)\npublic void testDoOptimizeSingular() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    // Since setTarget and setStartPoint methods do not exist, initialize the fields directly\n    // This requires making the fields accessible, which can be done by changing their access modifiers to protected\n    optimizertarget = target;\n    optimizer.startPoint = startPoint;\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.ConvergenceException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_2</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[53,18] target has private access in org.apache.commons.math3.optim.nonlinear.vector.MultivariateVectorOptimizer
[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[54,18] cannot find symbol
  symbol:   variable startPoint
  location: variable optimizer of type org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that doOptimize throws exception when internal error occurs', 'test_name': 'testDoOptimizeInternalError', 'test_code': '@Test(expected = MathInternalError.class)\npublic void testDoOptimizeInternalError() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            throw new MathInternalError();\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    // Since setTarget and setStartPoint methods do not exist, initialize the fields directly\n    // This requires making the fields accessible, which can be done by changing their access modifiers to protected\n    optimizer.target = target;\n    optimizer.startPoint = startPoint;\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathInternalError;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_2</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[53,18] lowerBound has private access in org.apache.commons.math3.optim.BaseMultivariateOptimizer
[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[54,18] upperBound has private access in org.apache.commons.math3.optim.BaseMultivariateOptimizer
[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[55,18] checkParameters() has private access in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that checkParameters throws exception when bounds are set', 'test_name': 'testCheckParametersBounds', 'test_code': '@Test(expected = MathUnsupportedOperationException.class)\npublic void testCheckParametersBounds() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] lowerBound = new double[1];\n    double[] upperBound = new double[1];\n    // Since setLowerBound and setUpperBound methods do not exist, initialize the fields directly\n    // This requires making the fields accessible, which can be done by changing their access modifiers to protected\n    optimizer.lowerBound = lowerBound;\n    optimizer.upperBound = upperBound;\n    optimizer.checkParameters();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathUnsupportedOperationException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_3</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullChecker(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.03 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the constructor throws a NullArgumentException when the convergence checker is null', 'test_name': 'testConstructorNullChecker', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullChecker() {\n    new GaussNewtonOptimizer((ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.NullArgumentException;\nimport org.apache.commons.math3.optim.ConvergenceChecker;\nimport org.apache.commons.math3.optim.PointVectorValuePair;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_3</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullCheckerBoolean(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.027 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the constructor throws a NullArgumentException when the convergence checker is null and useLU is specified', 'test_name': 'testConstructorNullCheckerBoolean', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullCheckerBoolean() {\n    new GaussNewtonOptimizer(true, (ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_3</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeSingular(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.039 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.ConvergenceException> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeSingular(GaussNewtonOptimizerTest.java:53)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the doOptimize method throws a ConvergenceException when the problem is singular', 'test_name': 'testDoOptimizeSingular', 'test_code': '@Test(expected = ConvergenceException.class)\npublic void testDoOptimizeSingular() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    // use reflection to set private fields\n    try {\n        java.lang.reflect.Field targetField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("target");\n        targetField.setAccessible(true);\n        targetField.set(optimizer, target);\n        java.lang.reflect.Field startPointField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("startPoint");\n        startPointField.setAccessible(true);\n        startPointField.set(optimizer, startPoint);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.ConvergenceException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_3</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeInternalError(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.028 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.MathInternalError> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeInternalError(GaussNewtonOptimizerTest.java:53)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the doOptimize method throws a MathInternalError when the convergence checker throws a MathInternalError', 'test_name': 'testDoOptimizeInternalError', 'test_code': '@Test(expected = MathInternalError.class)\npublic void testDoOptimizeInternalError() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            throw new MathInternalError();\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    // use reflection to set private fields\n    try {\n        java.lang.reflect.Field targetField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("target");\n        targetField.setAccessible(true);\n        targetField.set(optimizer, target);\n        java.lang.reflect.Field startPointField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("startPoint");\n        startPointField.setAccessible(true);\n        startPointField.set(optimizer, startPoint);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathInternalError;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_1_3</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testCheckParametersBounds(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.035 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.MathUnsupportedOperationException> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testCheckParametersBounds(GaussNewtonOptimizerTest.java:53)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the checkParameters method throws a MathUnsupportedOperationException when bounds are specified', 'test_name': 'testCheckParametersBounds', 'test_code': '@Test(expected = MathUnsupportedOperationException.class)\npublic void testCheckParametersBounds() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] lowerBound = new double[1];\n    double[] upperBound = new double[1];\n    // use reflection to set private fields\n    try {\n        java.lang.reflect.Field lowerBoundField = optimizer.getClass().getSuperclass().getDeclaredField("lowerBound");\n        lowerBoundField.setAccessible(true);\n        lowerBoundField.set(optimizer, lowerBound);\n        java.lang.reflect.Field upperBoundField = optimizer.getClass().getSuperclass().getDeclaredField("upperBound");\n        upperBoundField.setAccessible(true);\n        upperBoundField.set(optimizer, upperBound);\n        java.lang.reflect.Method checkParametersMethod = optimizer.getClass().getDeclaredMethod("checkParameters");\n        checkParametersMethod.setAccessible(true);\n        checkParametersMethod.invoke(optimizer);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathUnsupportedOperationException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-INFO">INFO</td>
                    <td>f_1</td>
                    <td>158.04768085479736</td>
                    <td>20733</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullChecker(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.04 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the GaussNewtonOptimizer constructor with a null convergence checker.\n', 'test_name': 'testConstructorNullChecker', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullChecker() {\n    new GaussNewtonOptimizer((ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.NullArgumentException;\nimport org.apache.commons.math3.optim.ConvergenceChecker;\nimport org.apache.commons.math3.optim.PointVectorValuePair;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullCheckerBoolean(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.028 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the GaussNewtonOptimizer constructor with a null convergence checker and useLU flag.\n', 'test_name': 'testConstructorNullCheckerBoolean', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullCheckerBoolean() {\n    new GaussNewtonOptimizer(true, (ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeSingular(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.034 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.ConvergenceException> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeSingular(GaussNewtonOptimizerTest.java:53)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the doOptimize method with a singular matrix.\n', 'test_name': 'testDoOptimizeSingular', 'test_code': '@Test(expected = ConvergenceException.class)\npublic void testDoOptimizeSingular() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    // use reflection to set private fields\n    try {\n        java.lang.reflect.Field targetField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("target");\n        targetField.setAccessible(true);\n        targetField.set(optimizer, target);\n        java.lang.reflect.Field startPointField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("startPoint");\n        startPointField.setAccessible(true);\n        startPointField.set(optimizer, startPoint);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.ConvergenceException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeInternalError(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.04 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.MathInternalError> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeInternalError(GaussNewtonOptimizerTest.java:53)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the doOptimize method with a MathInternalError.\n', 'test_name': 'testDoOptimizeInternalError', 'test_code': '@Test(expected = MathInternalError.class)\npublic void testDoOptimizeInternalError() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            throw new MathInternalError();\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    // use reflection to set private fields\n    try {\n        java.lang.reflect.Field targetField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("target");\n        targetField.setAccessible(true);\n        targetField.set(optimizer, target);\n        java.lang.reflect.Field startPointField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("startPoint");\n        startPointField.setAccessible(true);\n        startPointField.set(optimizer, startPoint);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathInternalError;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>g_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testCheckParametersBounds(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.031 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.MathUnsupportedOperationException> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testCheckParametersBounds(GaussNewtonOptimizerTest.java:53)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test the checkParameters method with bounds.\n', 'test_name': 'testCheckParametersBounds', 'test_code': '@Test(expected = MathUnsupportedOperationException.class)\npublic void testCheckParametersBounds() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] lowerBound = new double[1];\n    double[] upperBound = new double[1];\n    // use reflection to set private fields\n    try {\n        java.lang.reflect.Field lowerBoundField = optimizer.getClass().getSuperclass().getDeclaredField("lowerBound");\n        lowerBoundField.setAccessible(true);\n        lowerBoundField.set(optimizer, lowerBound);\n        java.lang.reflect.Field upperBoundField = optimizer.getClass().getSuperclass().getDeclaredField("upperBound");\n        upperBoundField.setAccessible(true);\n        upperBoundField.set(optimizer, upperBound);\n        java.lang.reflect.Method checkParametersMethod = optimizer.getClass().getDeclaredMethod("checkParameters");\n        checkParametersMethod.setAccessible(true);\n        checkParametersMethod.invoke(optimizer);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathUnsupportedOperationException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-INFO">INFO</td>
                    <td>g_2</td>
                    <td>45.23683524131775</td>
                    <td>5998</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullChecker(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.028 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the constructor throws a NullArgumentException when given a null ConvergenceChecker\n', 'test_name': 'testConstructorNullChecker\n', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullChecker() {\n    new GaussNewtonOptimizer((ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.NullArgumentException;\nimport org.apache.commons.math3.optim.ConvergenceChecker;\nimport org.apache.commons.math3.optim.PointVectorValuePair;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullCheckerBoolean(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.036 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the constructor throws a NullArgumentException when given a null ConvergenceChecker and a boolean value\n', 'test_name': 'testConstructorNullCheckerBoolean\n', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullCheckerBoolean() {\n    new GaussNewtonOptimizer(true, (ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': ''}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeSingular(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.029 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.ConvergenceException> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeSingular(GaussNewtonOptimizerTest.java:52)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the doOptimize method throws a ConvergenceException when the problem is singular\n', 'test_name': 'testDoOptimizeSingular\n', 'test_code': '@Test(expected = ConvergenceException.class)\npublic void testDoOptimizeSingular() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    try {\n        java.lang.reflect.Field targetField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("target");\n        targetField.setAccessible(true);\n        targetField.set(optimizer, target);\n        java.lang.reflect.Field startPointField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("startPoint");\n        startPointField.setAccessible(true);\n        startPointField.set(optimizer, startPoint);\n        java.lang.reflect.Field weightField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("weight");\n        weightField.setAccessible(true);\n        weightField.set(optimizer, new double[1][1]);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.ConvergenceException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_1</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeInternalError(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.027 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.MathInternalError> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeInternalError(GaussNewtonOptimizerTest.java:52)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the doOptimize method throws a MathInternalError when the convergence checker throws a MathInternalError\n', 'test_name': 'testDoOptimizeInternalError\n', 'test_code': '@Test(expected = MathInternalError.class)\npublic void testDoOptimizeInternalError() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            throw new MathInternalError();\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    try {\n        java.lang.reflect.Field targetField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("target");\n        targetField.setAccessible(true);\n        targetField.set(optimizer, target);\n        java.lang.reflect.Field startPointField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("startPoint");\n        startPointField.setAccessible(true);\n        startPointField.set(optimizer, startPoint);\n        java.lang.reflect.Field weightField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("weight");\n        weightField.setAccessible(true);\n        weightField.set(optimizer, new double[1][1]);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathInternalError;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_1</td>
                    <td>Compilation failure</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell">[ERROR] /home/sijia/test-generation-coverage/guided-test-generation/defects4j-subjects-notests/Math-2f/src/test/java/org/apache/commons/math3/optim/nonlinear/vector/jacobian/GaussNewtonOptimizerTest.java:[61,18] checkParameters() has private access in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer
</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that the checkParameters method throws a MathUnsupportedOperationException when bounds are provided\n', 'test_name': 'testCheckParametersBounds\n', 'test_code': '@Test(expected = MathUnsupportedOperationException.class)\npublic void testCheckParametersBounds() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] lowerBound = new double[1];\n    double[] upperBound = new double[1];\n    try {\n        java.lang.reflect.Field lowerBoundField = optimizer.getClass().getSuperclass().getDeclaredField("lowerBound");\n        lowerBoundField.setAccessible(true);\n        lowerBoundField.set(optimizer, lowerBound);\n        java.lang.reflect.Field upperBoundField = optimizer.getClass().getSuperclass().getDeclaredField("upperBound");\n        upperBoundField.setAccessible(true);\n        upperBoundField.set(optimizer, upperBound);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n    optimizer.checkParameters();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathUnsupportedOperationException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullChecker(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.031 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that a NullArgumentException is thrown when the convergence checker is null in the GaussNewtonOptimizer constructor.', 'test_name': 'testConstructorNullChecker', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullChecker() {\n    new GaussNewtonOptimizer((ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.NullArgumentException;\nimport org.apache.commons.math3.optim.ConvergenceChecker;\nimport org.apache.commons.math3.optim.PointVectorValuePair;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullCheckerBoolean(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.037 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that a NullArgumentException is thrown when the convergence checker is null in the GaussNewtonOptimizer constructor with boolean parameter.', 'test_name': 'testConstructorNullCheckerBoolean', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullCheckerBoolean() {\n    new GaussNewtonOptimizer(true, (ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeSingular(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.032 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.ConvergenceException> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeSingular(GaussNewtonOptimizerTest.java:52)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that a ConvergenceException is thrown when the linearized least squares problem is singular.', 'test_name': 'testDoOptimizeSingular', 'test_code': '@Test(expected = ConvergenceException.class)\npublic void testDoOptimizeSingular() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    try {\n        java.lang.reflect.Field targetField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("target");\n        targetField.setAccessible(true);\n        targetField.set(optimizer, target);\n        java.lang.reflect.Field startPointField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("startPoint");\n        startPointField.setAccessible(true);\n        startPointField.set(optimizer, startPoint);\n        java.lang.reflect.Field weightField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("weight");\n        weightField.setAccessible(true);\n        weightField.set(optimizer, new double[1][1]);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.ConvergenceException;'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeInternalError(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.033 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.MathInternalError> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeInternalError(GaussNewtonOptimizerTest.java:52)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that a MathInternalError is thrown when the convergence checker throws a MathInternalError.', 'test_name': 'testDoOptimizeInternalError', 'test_code': '@Test(expected = MathInternalError.class)\npublic void testDoOptimizeInternalError() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            throw new MathInternalError();\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    try {\n        java.lang.reflect.Field targetField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("target");\n        targetField.setAccessible(true);\n        targetField.set(optimizer, target);\n        java.lang.reflect.Field startPointField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("startPoint");\n        startPointField.setAccessible(true);\n        startPointField.set(optimizer, startPoint);\n        java.lang.reflect.Field weightField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("weight");\n        weightField.setAccessible(true);\n        weightField.set(optimizer, new double[1][1]);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathInternalError;'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_2</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testCheckParametersBounds(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.038 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.MathUnsupportedOperationException> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testCheckParametersBounds(GaussNewtonOptimizerTest.java:52)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that a MathUnsupportedOperationException is thrown when bounds are passed to the optimize method.', 'test_name': 'testCheckParametersBounds', 'test_code': '@Test(expected = MathUnsupportedOperationException.class)\npublic void testCheckParametersBounds() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] lowerBound = new double[1];\n    double[] upperBound = new double[1];\n    try {\n        java.lang.reflect.Field lowerBoundField = optimizer.getClass().getSuperclass().getDeclaredField("lowerBound");\n        lowerBoundField.setAccessible(true);\n        lowerBoundField.set(optimizer, lowerBound);\n        java.lang.reflect.Field upperBoundField = optimizer.getClass().getSuperclass().getDeclaredField("upperBound");\n        upperBoundField.setAccessible(true);\n        upperBoundField.set(optimizer, upperBound);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathUnsupportedOperationException;'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_3</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullChecker(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.038 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that a NullArgumentException is thrown when the convergence checker is null in the GaussNewtonOptimizer constructor', 'test_name': 'testConstructorNullChecker', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullChecker() {\n    new GaussNewtonOptimizer((ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.NullArgumentException;\nimport org.apache.commons.math3.optim.ConvergenceChecker;\nimport org.apache.commons.math3.optim.PointVectorValuePair;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_3</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testConstructorNullCheckerBoolean(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.039 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: org.apache.commons.math3.exception.NullArgumentException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:35)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that a NullArgumentException is thrown when the convergence checker is null in the GaussNewtonOptimizer constructor with a boolean parameter', 'test_name': 'testConstructorNullCheckerBoolean', 'test_code': '@Test(expected = NullArgumentException.class)\npublic void testConstructorNullCheckerBoolean() {\n    new GaussNewtonOptimizer(true, (ConvergenceChecker<PointVectorValuePair>) null);\n}\n', 'new_imports_code': None}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_3</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeSingular(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.033 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.ConvergenceException> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeSingular(GaussNewtonOptimizerTest.java:52)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that a ConvergenceException is thrown when the linearized least squares problem is singular', 'test_name': 'testDoOptimizeSingular', 'test_code': '@Test(expected = ConvergenceException.class)\npublic void testDoOptimizeSingular() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    try {\n        java.lang.reflect.Field targetField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("target");\n        targetField.setAccessible(true);\n        targetField.set(optimizer, target);\n        java.lang.reflect.Field startPointField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("startPoint");\n        startPointField.setAccessible(true);\n        startPointField.set(optimizer, startPoint);\n        java.lang.reflect.Field weightField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("weight");\n        weightField.setAccessible(true);\n        weightField.set(optimizer, new double[1][1]);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.ConvergenceException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_3</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testDoOptimizeInternalError(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.032 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.MathInternalError> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testDoOptimizeInternalError(GaussNewtonOptimizerTest.java:52)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that a MathInternalError is thrown when the convergence checker throws a MathInternalError', 'test_name': 'testDoOptimizeInternalError', 'test_code': '@Test(expected = MathInternalError.class)\npublic void testDoOptimizeInternalError() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            throw new MathInternalError();\n        }\n    });\n    double[] target = new double[1];\n    double[] startPoint = new double[1];\n    try {\n        java.lang.reflect.Field targetField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("target");\n        targetField.setAccessible(true);\n        targetField.set(optimizer, target);\n        java.lang.reflect.Field startPointField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("startPoint");\n        startPointField.setAccessible(true);\n        startPointField.set(optimizer, startPoint);\n        java.lang.reflect.Field weightField = optimizer.getClass().getSuperclass().getSuperclass().getDeclaredField("weight");\n        weightField.setAccessible(true);\n        weightField.set(optimizer, new double[1][1]);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathInternalError;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-FAIL">FAIL</td>
                    <td>f_2_3</td>
                    <td>Test failures</td>
                    <td>1</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td><pre><code class="language-shell"> - in org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest
testCheckParametersBounds(org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest)  Time elapsed: 0.034 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<org.apache.commons.math3.exception.MathUnsupportedOperationException> but was<java.lang.RuntimeException>
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizerTest.testCheckParametersBounds(GaussNewtonOptimizerTest.java:52)</code></pre></td>
                    <td><pre><code class="language-python">{'test_behavior': 'Test that a MathUnsupportedOperationException is thrown when bounds are passed to the optimize method', 'test_name': 'testCheckParametersBounds', 'test_code': '@Test(expected = MathUnsupportedOperationException.class)\npublic void testCheckParametersBounds() {\n    GaussNewtonOptimizer optimizer = new GaussNewtonOptimizer(new ConvergenceChecker<PointVectorValuePair>() {\n        @Override\n        public boolean converged(int iteration, PointVectorValuePair previous, PointVectorValuePair current) {\n            return false;\n        }\n    });\n    double[] lowerBound = new double[1];\n    double[] upperBound = new double[1];\n    try {\n        java.lang.reflect.Field lowerBoundField = optimizer.getClass().getSuperclass().getDeclaredField("lowerBound");\n        lowerBoundField.setAccessible(true);\n        lowerBoundField.set(optimizer, lowerBound);\n        java.lang.reflect.Field upperBoundField = optimizer.getClass().getSuperclass().getDeclaredField("upperBound");\n        upperBoundField.setAccessible(true);\n        upperBoundField.set(optimizer, upperBound);\n    } catch (Exception e) {\n        throw new RuntimeException(e);\n    }\n    optimizer.doOptimize();\n}\n', 'new_imports_code': 'import org.apache.commons.math3.exception.MathUnsupportedOperationException;\n'}</code></pre></td>
                </tr>
                
                <tr>
                    <td class="status-INFO">INFO</td>
                    <td>f_2</td>
                    <td>164.29137301445007</td>
                    <td>22888</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                </tr>
                
                <tr>
                    <td class="status-INFO">INFO</td>
                    <td></td>
                    <td></td>
                    <td>0</td>
                    <td>0.0</td>
                    <td>0.0</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                </tr>
                
            </table>
            <script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.23.0/prism.min.js"></script>
        </body>
        </html>
        